docs/src/explanations/naming.md

```diff
@@ -202,6 +226,84 @@ class Example8 extends Module {
 ChiselStage.emitVerilog(new Example8)
 ```
 
+Note that using `.suggestName` does *not* affect prefixes derived from val names;
+however, it *can* affect prefixes derived from connections (eg. `:=`):
+
+```scala mdoc
+class ConnectionPrefixExample extends Module {
+  val in0 = IO(Input(UInt(2.W)))
+  val in1 = IO(Input(UInt(2.W)))
+
+  val out0 = IO(Output(UInt()))
+  val out1 = IO(Output(UInt()))
+
+  out0 := {
+    // out0_sum
+    val sum = in0 + in1
+    sum + 1.U
+  }
+  // Comes after so does *not* affect prefix above
+  out0.suggestName("foo")
+
+  // Comes before so *does* affect prefix below
+  out1.suggestName("bar")
+  out1 := {
+    // bar_diff
+    val diff = in0 - in1
```

Comment:
but what happens if I `suggestName("mySuggestedName")` on `diff`? Is it `out_mySuggestedName` or just `mySuggestedName`?

TBH I lost track of what it actually does, then it's a question is that a bug or a feature

Reply:
This is [already documented](https://www.chisel-lang.org/chisel3/docs/explanations/naming.html#suggest-a-signals-name-or-the-instance-name-of-a-module)

> If you want to specify the name of a signal, you can always use the .suggestName API. Please note that the suggested name will still be prefixed (including by the plugin). You can always use the noPrefix object to strip this.